### PR TITLE
ROX-13608 tunable thread table max size

### PR DESCRIPTION
**Pull request by ovalenti**
## Description

New environment variable (`ROX_COLLECTOR_MAX_THREAD_TABLE_SIZE`) which, when set, defines an upper bound to the thread table size of Falco.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

### CODE REVIEW

**Review by Molter73 — APPROVED** [review]
LGTM! Just left a comment on an avoidable string copy and a user facing variable

File: docs/references.md [hunk 1]
```diff
@@ -29,6 +29,13 @@ false.
 about the originator process on all network listening-endpoint objects.
 The default is false.
 
+* `ROX_FALCO_MAX_THREAD_TABLE_SIZE`: Prevent the Falco thread-table (list of
```
[the remark this answers is not in the document]
Comment (ovalenti) [reply docs/references.md]: It makes sense, I renamed as suggested; thanks !

**Comment by ovalenti** [comment]
It looks like the case when the thread-table grows over the limit is not really handled in falco.

https://github.com/stackrox/falcosecurity-libs/blob/754bc65b9c569bef07d7a23f88c27f53b316c2a4/userspace/libsinsp/sinsp.cpp#L886

**Comment by erthalion** [comment]
> It looks like the case when the thread-table grows over the limit is not really handled in falco.
> 
> https://github.com/stackrox/falcosecurity-libs/blob/754bc65b9c569bef07d7a23f88c27f53b316c2a4/userspace/libsinsp/sinsp.cpp#L886

It is, they consciously drop events:

https://github.com/stackrox/falcosecurity-libs/blob/754bc65b9c569bef07d7a23f88c27f53b316c2a4/userspace/libsinsp/threadinfo.cpp#L1394

**Comment by ovalenti** [comment]
> It is, they consciously drop events:
> 
> https://github.com/stackrox/falcosecurity-libs/blob/754bc65b9c569bef07d7a23f88c27f53b316c2a4/userspace/libsinsp/threadinfo.cpp#L1394

Absolutely, but then, when `add_thread()` returns _false_, the code does `ASSERT(false)`. Also, in that case, I later get a crash in falco.

I think that I will have to make sure that Falco handles this correctly before the max size of thread-table is interesting to set by a Stackrox user.

**Comment by erthalion** [comment]
> > It is, they consciously drop events:
> > https://github.com/stackrox/falcosecurity-libs/blob/754bc65b9c569bef07d7a23f88c27f53b316c2a4/userspace/libsinsp/threadinfo.cpp#L1394
> 
> Absolutely, but then, when `add_thread()` returns _false_, the code does `ASSERT(false)`. Also, in that case, I later get a crash in falco.

AFAIK asserts are ignored in the release mode, so it will be silenced. That's why we have so much troubles enabling debug mode for CI.

**Pull request closed without merging**